pkg/client/restclient/config.go

```diff
@@ -276,11 +275,16 @@ func InClusterConfig() (*Config, error) {
 // Note: the Insecure flag is ignored when testing for this value, so MITM attacks are
 // still possible.
 func IsConfigTransportTLS(config Config) bool {
-	baseURL, _, err := defaultServerUrlFor(&config)
+	hosts, _, err := defaultServerUrlFor(&config)
```

Comment:
Agree. Normal transport building shouldn't allow combining credentials with non-https URLs. I hope that's still the case.